osu.Game/Screens/Edit/Timing/TapButton.cs

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Framework.Threading;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Input.Bindings;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Edit.Timing
+{
+    internal class TapButton : CircularContainer, IKeyBindingHandler<GlobalAction>
+    {
+        public const float SIZE = 140;
+
+        public readonly BindableBool IsHandlingTapping = new BindableBool();
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        [Resolved(canBeNull: true)]
+        private Bindable<ControlPointGroup>? selectedGroup { get; set; }
+
+        [Resolved(canBeNull: true)]
+        private IBeatSyncProvider? beatSyncSource { get; set; }
+
+        private Circle hoverLayer = null!;
+
+        private CircularContainer innerCircle = null!;
+        private Box innerCircleHighlight = null!;
+
+        private int currentLight;
+
+        private Container scaleContainer = null!;
+        private Container lights = null!;
+        private Container lightsGlow = null!;
+        private OsuSpriteText bpmText = null!;
+        private Container textContainer = null!;
+
+        private bool grabbedMouseDown;
+
+        private ScheduledDelegate? resetDelegate;
+
+        private const int light_count = 8;
+
+        private const int initial_taps_to_ignore = 4;
+
+        private const int max_taps_to_consider = 16;
```

Comment:
I find this way too small of an amount to stabilise with. I'm going to bring up my reference of https://www.beatsperminuteonline.com/, which doesn't reset while you're continuously pressing it, and say that I've never had any issues stabilising that.

1 BPM makes a huge difference, especially on longer maps where you can't really tell visually how much you're off by. This is a perfectly offset timing point, but 1 BPM off - I can't tell at all even if I scroll through the waveforms:
![image](https://user-images.githubusercontent.com/1329837/171586147-d5b178c7-8372-4c49-9d97-c47f33ee009d.png)

So that's why I'd like for this to be removed or increased to some value where you're unlikely to hit it unless you're bored, like 128.

Reply:
The idea is to avoid a scenario where the user changes tapping speed at some point (ie. if bpm is variable or they decide to double their tap speed). I'll lift it for now to avoid contention, but will implement a better algorithm that will handle such changes in tapping speed in the future.